controller/src/main/java/org/jboss/as/controller/DefaultAttributeMarshaller.java

```diff
@@ -32,13 +32,17 @@
  */
 public class DefaultAttributeMarshaller extends AttributeMarshaller {
 
-
+    @Override
     public void marshallAsAttribute(final AttributeDefinition attribute, final ModelNode resourceModel, final boolean marshallDefault, final XMLStreamWriter writer) throws XMLStreamException {
         if (isMarshallable(attribute, resourceModel, marshallDefault)) {
-            writer.writeAttribute(attribute.getXmlName(), resourceModel.get(attribute.getName()).asString());
+            writer.writeAttribute(attribute.getXmlName(), this.asString(resourceModel.get(attribute.getName())));
         }
     }
 
+    /**
+     * Use {@link #marshallAsElement(AttributeDefinition, ModelNode, boolean, XMLStreamWriter)} instead.
+     */
+    @Deprecated
```

Comment:
I was going to object to deprecating; but you're right. It's not in the AttributeMarshaller interface. Good call.